src/sciline/handler.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from typing import Callable, NoReturn, Protocol, Type, TypeVar, Union
+
+from .typing import Item
+
+T = TypeVar('T')
+
+
+class UnsatisfiedRequirement(Exception):
+    """Raised when a type cannot be provided."""
+
+
+class Handler(Protocol):
```

Comment:
```suggestion
class ErrorHandler(Protocol):
```
And rename the file.

Or pick a different name. But 'handler' is too generic.